### ISSUE TITLE
Unable to chain AVP to existing helm tool 

### ISSUE THREAD

**Issue by borisovano** [argoproj-labs/argocd-vault-plugin#579]
Instead of chaining the plugin to the output of helm tool which is minimal and elegant. The user is required to implement all the helm heavy lifting by himself.

So instead of using the output of helm tool we need to use on of the following:
Default: https://argocd-vault-plugin.readthedocs.io/en/stable/installation/#initcontainer-and-configuration-via-sidecar
Helm Simple: https://argocd-vault-plugin.readthedocs.io/en/stable/usage/#with-additional-helm-arguments
Helm with arguments: https://argocd-vault-plugin.readthedocs.io/en/stable/usage/#with-additional-helm-arguments
Helm with inline values: https://argocd-vault-plugin.readthedocs.io/en/stable/usage/#with-an-inline-values-file
Helm with value files from another repo (multisource): No docs

(PS - it's not clear from the docs which usecase fits each direction)

Some of the directions above have security risks, others prone to errors and none allows me to keep my application as is.
So instead of current implementation, the application is as follows:

```
project: myproj
source:
  repoURL: 'https://gitlab.com/myrepo'
  path: local-charts/mychart
  targetRevision: HEAD
  helm:
    valueFiles:
      - ../../services/myservice/myapp/values.yaml
      - >-
        ../../services/myservice/myapp/env/myenv/values.yaml
```
The user required to use something like this:

```
  source:
    path: your-app
    plugin:
      name: argocd-vault-plugin-helm
      env:
        - name: HELM_VALUES
          value: |
            # non-vault helm values are specified normally
            someValue: lasldkfjlksa
            moreStuff:
              - a
              - b
              - c
```

Proposal
CMP plugin should get the helm tool output and return valid manifests

```
project: myproj
source:
  repoURL: 'https://gitlab.com/myrepo'
  path: local-charts/mychart
  targetRevision: HEAD
  helm:
    valueFiles:
      - ../../services/myservice/myapp/values.yaml
      - >-
        ../../services/myservice/myapp/env/myenv/values.yaml
   plugin: 
      name: argo-vault-plugin
      after: helm
```

**Comment by werne2j** [comment]
This is not possible. It’s an issue with Argo CD. AVP is just a binary that Argo CD calls as a CMP. 

If this is a feature you’d like, you would need to create a feature request in Argo CD.

**Comment by borisovano** [comment]
Already done, thank you 🙏 

PS - https://github.com/argoproj/argo-cd/issues/16252
